lmnet/lmnet/datasets/ytfaces.py

```diff
@@ -0,0 +1,95 @@
+# -*- coding: utf-8 -*-
+# Copyright 2018 The Blueoil Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# =============================================================================
+import os.path
+
+import numpy as np
+
+from lmnet.utils.image import load_image
+from lmnet.datasets.base import KeypointDetectionBase
+
+
+class YoutubeFacialLandmarks(KeypointDetectionBase):
+    """Youtube Facial landmarks detection dataset. This dataset is taken from:
+    https://github.com/jerrychen44/face_landmark_detection_pytorch
+
+    """
+
+    classes = ['face']
+    num_classes = len(classes)
+    available_subsets = ["train", "validation"]
+    extend_dir = "ytfaces"
+
+    def __init__(self, subset="train", batch_size=10, *args, **kwargs):
+        super().__init__(subset=subset, batch_size=batch_size, *args, **kwargs)
+
+        if subset == 'train':
+            self.csv = os.path.join(self.data_dir, "training_frames_keypoints.csv")
+            self.image_dir = os.path.join(self.data_dir, "training")
+        elif subset == 'validation':
+            self.csv = os.path.join(self.data_dir, "test_frames_keypoints.csv")
+            self.image_dir = os.path.join(self.data_dir, "test")
+
+        self.num_joints = 68
+
+        self.files, self.joints_list = self._load_csv()
+
+    def _load_csv(self):
+        """Read items from JSON files"""
+        files = []
+        joints_list = []
+        num_dimensions = 2
+
+        with open(self.csv) as f:
+            # skip header line
+            f.readline()
+            for line in f:
+                landmarks = line.split(",")
```

Comment:
`filename, *landmarks = line.split(",")`

Reply:
This is elegant. 👍